hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleStakingInfo.java

```diff
@@ -233,4 +281,21 @@ public void setRewardSumHistory(final long[] rewardSumHistory) {
 	public void clearRewardSumHistory() {
 		Arrays.fill(rewardSumHistory, 0);
 	}
+
+	public void updateRewardSumHistory(final double rewardRate, final long totalStakedRewardStart) {
+		rewardSumHistory = Arrays.copyOf(rewardSumHistory, rewardSumHistory.length);
+		final var droppedRewardSum = rewardSumHistory[rewardSumHistory.length-1];
+		for (int i = rewardSumHistory.length-1; i > 0; i--) {
+			rewardSumHistory[i] = rewardSumHistory[i - 1] - droppedRewardSum;
+		}
+		rewardSumHistory[0] -= droppedRewardSum;
+			/*
+				if this node was "active", then it should give rewards for this staking period
+				if (node.numRoundsWithJudge / numRoundsInPeriod >= activeThreshold)
+			 */
+		rewardSumHistory[0] += totalStakedRewardStart == 0 ? 0 :
+				(long) (rewardRate * stakeRewardStart / totalStakedRewardStart / 100_000_000);
```

Comment:
We could define a Services equivalent of [`Units`](https://github.com/swirlds/swirlds-platform/blob/develop/swirlds-common/src/main/java/com/swirlds/common/utility/Units.java) and reference `HBARS_TO_TINYBARS` here instead of `100_000_000`.